plugins/normalize-whitespace/prism-normalize-whitespace.js

```diff
@@ -161,6 +173,29 @@
 			return;
 		}
 
+		if (env.settings == null) { env.settings = {}; }
+
+		// Read settings from 'data-' attributes
+		for (var key in settingsConfig) {
+			if (Object.hasOwnProperty.call(settingsConfig, key)) {
+				var settingType = settingsConfig[key];
+				if (pre.hasAttribute('data-' + key)) {
+					var value = pre.getAttribute('data-' + key);
+					switch (settingType) {
+						case 'boolean':
+							value = (value === 'true' || value === '') || (value === 'false' ? false : undefined);
+							break;
+						case 'number':
+							value = (value === '' || (isNaN(Number(value))) ? undefined : Number(value));
+							break;
+					}
+					if (value !== undefined) {
+						env.settings[key] = value;
+					}
```

Comment:
Actually, how about this?
```suggestion
					try {
						var value = JSON.parse(pre.getAttribute('data-' + key) || 'true')
						if (typeof value === settingType) {
							env.settings[key] = value;
						}
					} catch (_error) {
						// ignore error
					}
```

Reply:
> if that matters to you

It doesn't. Highlighting and the normalize plugin itself will take orders of magnitude longer. There's no need to worry about this.